docs/plugin_developers_guide/01_Fledge_plugins.rst

```diff
@@ -41,11 +41,109 @@ Plugins
+- **Filter Plugins** - Filter plugins are used to modify data as it flows through Fledge. One or more filter plugins may compose a pipeline which modifies, either, data flowing out from the South ingress tasks into the Fledge Storage system, or out from Fledge storage into the North egress tasks.
+- **Notification Rule Plugins** - Notification plugins evaluate data after it has been entered into the Fledge Storage, before it enters the northbound Filter pipeline(s). If the notification evaluates to True, a rule associated with the condition is executed. (Notification rules plugins require installation of the optional notification service.)
+- **Notification Delivery Plugins** - These plugins deliver a notification event; events, trigger "reason", and accompanying "message" are specified, and delivered to a given delivery channel (eg., email/sms/...). (Notification delivery plugins require installation of the optional notification service.)
+- **North Plugins** - These plugins take data, originated by South service(s), optionally transformed by filters, optionally triggering notifications, and export those data to the outside world.
+
+Fledge message contents
+=======================
+Information flows through fledge in "messages". Messages have a few "tags" which have well defined values. Messages may be extended to contain additional tags appropriate to particular plugins and data flows.
+
+Most messages include:
+
++------------+------------------+-------------------------------------------------------------------------------------------------------+
+| Tag        | Type             | Description                                                                                           |
++============+==================+=======================================================================================================+
+| asset      | string           | Name under which this stream of data are placeed in Storage                                           |
++------------+------------------+-------------------------------------------------------------------------------------------------------+
+| timestamp  | string           | Fledge timestamp when this asset was generated                                                        |
++------------+------------------+-------------------------------------------------------------------------------------------------------+
+| key        | string           | unique ID                                                                                             |
++------------+------------------+-------------------------------------------------------------------------------------------------------+
+| readings   | dict             | {"<name-1>": <value-1> [, "<name-2>": <value-2>,... ["user_ts": <device-timestamp>]]}                 |
```

Comment:
There is a lot more to this as it represents the actual data, we should probably tell the reader that vital bit of information.